### PR TITLE
JAMES-2586 Add search module chooser for Postgres app

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

File: server/apps/postgres-app/pom.xml [hunk 1]
```diff
@@ -151,6 +179,12 @@
             <groupId>${james.groupId}</groupId>
             <artifactId>james-server-guice-webadmin-mailrepository</artifactId>
         </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-jmap-draft-integration-testing</artifactId>
```
Comment (chibenwa) [inline server/apps/postgres-app/pom.xml:184]: Question: what is this needed for?

File: server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/LuceneSearchMailboxModule.java [hunk 2]
```diff
@@ -1,58 +0,0 @@
...
-import org.apache.lucene.store.Directory;
-import org.apache.lucene.store.FSDirectory;
-
-import com.google.inject.AbstractModule;
-import com.google.inject.Provides;
-import com.google.inject.Scopes;
-import com.google.inject.Singleton;
-import com.google.inject.multibindings.Multibinder;
-
-public class LuceneSearchMailboxModule extends AbstractModule {
```
Comment (chibenwa) [inline server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/LuceneSearchMailboxModule.java:38]: Please remove the Lucene dependency, then!

File: server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/PostgresMailboxModule.java [hunk 3]
```diff
@@ -85,7 +85,6 @@ protected void configure() {
         postgresDataDefinitions.addBinding().toInstance(PostgresMailboxAggregateModule.MODULE);
 
         install(new PostgresQuotaModule());
-        install(new PostgresQuotaSearchModule());
```
Comment (chibenwa) [inline server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/PostgresMailboxModule.java:88]: Could you confirm quota search is well implemented with OpenSearch?

**Review by chibenwa — COMMENTED** [review]
Nice work.

I want it to be easy to start this. Please add:

 - Folder `sample-configuration-distributed` with sample file `opensearch.properties`
 - Sample docker compose `docker-compose-distributed.yml`
 - A line in the README on how to start that  `docker-compose-distributed.yml`

File: server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/PostgresMailboxModule.java (hunk 3 above)
Comment (Arsnael) [reply server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/PostgresMailboxModule.java:88]: It is cause when I had this, test were failing on a double injection?

Comment (Arsnael) [reply server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/PostgresMailboxModule.java:88]: To be a bit more precise they are being injected by the module chooser

File: server/apps/postgres-app/src/test/java/org/apache/james/PostgresWithTikaTest.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    static JamesServerExtension jamesServerExtension = new JamesServerBuilder<PostgresJamesConfiguration>(tmpDir ->
+        PostgresJamesConfiguration.builder()
+            .workingDirectory(tmpDir)
+            .configurationFromClasspath()
+            .searchConfiguration(SearchConfiguration.openSearch())
+            .usersRepository(DEFAULT)
+            .build())
+        .server(PostgresJamesServerMain::createServer)
+        .extension(new DockerOpenSearchExtension())
+        .extension(new TikaExtension())
```
Comment (vttranlina) [inline server/apps/postgres-app/src/test/java/org/apache/james/PostgresWithTikaTest.java:40]: Q: Will we support tika in Postgres?

(just re-confirm)

**Review by vttranlina — APPROVED** [review]

File: server/apps/postgres-app/src/test/java/org/apache/james/PostgresWithTikaTest.java (hunk 4 above)
Comment (Arsnael) [reply server/apps/postgres-app/src/test/java/org/apache/james/PostgresWithTikaTest.java:40]: Some injections with the search module chooser required the tika module. So I had to test it. But if not tika properties file, it is disabled and we use jsoupextractor instead. From what I understand when checking quickly

**Review by quantranhong1999 — APPROVED** [review]

**Comment by Arsnael** [comment]
Added the docker compose distributed with openSearch.

Also while I was at it, reworked the readme to remove all JPA mentions, propose use of postgres app with only postgres or distributed (opensearch), rework the normal docker compose to have scanning conf instead of opensearch

**Review by quantranhong1999 — APPROVED** [review]

File: server/apps/postgres-app/README.adoc [hunk 5]
```diff
@@ -26,15 +59,27 @@ Once everything is set up, you just have to run the jar with:
...
 Note that binding ports below 1024 requires administrative rights.
 
+=== Running distributed
+
+If you want to use the distributed version of James Postgres app, you will need to add configuration in the **conf** folder like in the
+[sample directory](https://github.com/apache/james-project/tree/master/server/apps/postgres-app/sample-configuration-distributed).
+
+Then you need to generate the keystore, rebuild the application jar and run it like above.
+
 == Docker distribution
```
Comment (chibenwa) [inline server/apps/postgres-app/README.adoc:83]: IMO we can remove this docker section and just replace it by docker-compose.

File: server/apps/postgres-app/sample-configuration-single/search.properties [hunk 6]
```diff
@@ -0,0 +1 @@
+implementation=scanning
```
Comment (chibenwa) [inline server/apps/postgres-app/sample-configuration-single/search.properties:1]: Add a comment: `not for production purposes. To be replaced by PG based search.`

**Review by chibenwa — APPROVED** [review]

**Comment by Arsnael** [comment]
Squashed and rebased

**Pull request merged**